doc/development/guide/documentation.rst

```diff
@@ -1,39 +1,455 @@
+
+        .. UsageDetails::
+
+            More complicated use cases, options, and larger code blocks (optional).
+
+        **Related tutorials**
+
+        Links to any relevant PennyLane demos (optional).
+        """
+
+Docstrings for classes follow a similar structure as functions, with a few differences:
+
+* **Include initialization arguments.** The constructor ``__init__()`` is documented
+  here. Note that as the constructor simply returns an instance of the class, no ``Returns:``
+  section is included.
+
+* **Do not list methods and properties.** Class properties and methods will be automatically
+  listed alongside the class in the generated documentation.
+
+* **List attributes if relevant.**
+
+Document all methods and properties with docstrings below the method declaration,
+as you would for functions above. However, there are two
+exemptions, where docstrings should *not* be provided:
+
+* **Magic or special methods.** This includes methods such as ``__init__``, ``__call__``,
+  ``__len__``, ``__getitem__``, etc. Only provide docstrings for special methods if
+  their behaviour is modified in a non-standard way (e.g., if negative indices are allowed
+  as arguments for ``__getitem__``).
+
+* **Overwritten methods.** Overwritten inherited methods automatically inherit the parent methods
+  docstring, even when overwritten. This is particularly useful when creating an instance
+  of an abstract base class; the abstract method docstring is defined in the parent, and automatically
+  inherited by the child class, even when overwritten.
+
+  As with special methods, only provide docstrings for overwritten methods if
+  their behaviour is modified in a non-standard way (as defined in the parent class).
+
+
+Variables
+~~~~~~~~~
+
+Module-level variables may also be optionally documented, by providing a triple-quote docstring
+**below** the variable definition. For example,
+
+.. code-block:: python
+
+    x = {"John": 23, "James": 54}
+    """dict[str, int]: stores the ages of known users"""
+
+The syntax is the same as those used for describing function arguments.
+
+
+.. _code_examples:
+
+Code examples
+-------------
+
+Code examples are very important; they *show* readers how the function or class should be used.
+When writing code examples for docstrings, use the following guidelines:
+
+- You may assume that PennyLane is imported as ``qml`` and NumPy is imported as ``np`` in the code examples.
+  All other imports must be specified explicitly.
+
+- For single line statements and associated output, use Python console syntax (``pycon``):
+
+  .. code-block:: rest
+
+      >>> circuit(0.5, 0.1)
+      [0.43241234312, -0.543534534]
+
+  Multi-line statements should use ``...`` to indicate continuation lines:
+
+  .. code-block:: rest
+
+      >>> dev = qml.device("default.qubit", wires=1)
+      >>> @qml.qnode(dev)
+      >>> def circuit(x):
+      ...     qml.RX(x, wires=0)
+      ...     return qml.expval(qml.PauliZ(0))
```

Comment:
These two are rendered in grey and not in black as the rest of the code (as if they were comments). Could this be because of the `.. code-block:: rest`? Shouldn't this be a `python3` block?

Reply:
This should be `rest`, since the code-block is showing the excerpt from a reStructuredText document. Unfortunately, it looks like the pygments `rest` lexer is not smart enough to know that `>>>` and `...` are valid rest syntax 😢

Reply:
I've changed it to `.. code-block:: pycon`, even though it's not ideal :(